assets/js/date-formatter.js

```diff
@@ -0,0 +1,112 @@
+const inputDate = document.getElementById('date');
+const inputDateFormat = document.getElementById('input-date-format');
+const outputDateFormat = document.getElementById('output-date-format');
+const formatButton = document.getElementById('format-button');
+const resultDateForm = document.getElementById('result-date-form');
+const fromNowButton = document.getElementById('from-now-button');
+const resultDateFromNow = document.getElementById('result-date-from-now');
+
+const dateFormats = [
+    'YYYYMMDD',
+    'DDMMYYYY',
+    'DD-MM-YYYY',
+    'YYYY-MM-DD',
+    'YYYY.MM.DD',
+    'DD.MM.YYYY',
+    'DD MM YYYY',
+    'YYYY MM DD'
+]
+
+dateFormats.forEach(el => {
+    const optionInput = document.createElement('option');
+    const optionOutput = document.createElement('option');
+    optionInput.value = el;
+    optionOutput.value = el;
+    optionInput.textContent = el;
+    optionOutput.textContent = el;
+    inputDateFormat.appendChild(optionInput);
+    outputDateFormat.appendChild(optionOutput);
+});
+
+
+let dateValue = '';
+
+inputDate.onchange = () => {
+    dateValue = inputDate.value;
+}
+
+formatButton.addEventListener('click', () => {
+    resultDateForm.textContent = dateFormatter(dateValue, inputDateFormat.value, outputDateFormat.value);
+})
+
+function deleteSeparators(dateException) {
+    if (dateException.match(/[^\d]/g)) {
+        console.log('here');
+        return dateException.replaceAll(/[^\d]/g, '');
+    }
+    return dateException;
+}
+
+const dateFormatter = (dateException, inputRegFormat = 'DDMMYYYY', outputRegFormat = 'DD-MM-YYYY') => {
+    dateException = deleteSeparators(dateException);
+    let separator = '';
+    if (dateException.length === 8 && /^\d+$/.test(dateException)) {
+        let resultDateForm = {};
+        if (/^DD/.test(inputRegFormat)) {
```

Comment:
_Code maintainability_: The date formatter expects "MM" always be in the middle. But some popular date formats ("MM/DD/YYYY" in the USA) put month to the start. If someone will add this format to the `dateFormats` array, you'll need to rework all the function. Formatter shouldn't rely on days/months/years position implicitly. Try to look to the usage of `str.exec()` or `str.indexOf()`. This note also applies to the output formatting.

Reply:
fixed